specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/virtualNetworkGateway.json

```diff
@@ -2214,20 +2214,30 @@
       ],
       "description": "The virtual network connection reset shared key"
     },
-    "ConnectionSharedKey": {
+    "ConnectionSharedKeyPropertiesFormat": {
       "properties": {
         "value": {
           "type": "string",
-          "description": "The virtual network connection shared key value."
+          "description": "The virtual network gateway connection shared key value."
+        }
+      },
+      "description": "Properties of the virtual network gateway connection shared key."
+    },
+    "ConnectionSharedKey": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ConnectionSharedKeyPropertiesFormat",
+          "description": "Properties of the virtual network gateway connection shared key."
         }
       },
-	  "allOf": [
+      "allOf": [
         {
           "$ref": "./network.json#/definitions/SubResource"
         }
       ],
       "required": [
-        "value"
+        "properties"
```

Comment:
won't it be a breaking change? please sync up with swagger team @dsgouda and need to review won't it break PS?

Reply:
@shahabhijeet does changing required properties cause a breaking change?
Will try to generate the SDKs and confirm at my end.

Reply:
@yushwang @MikhailTryakhov confirmed these will be breaking changes in the SDK. If this is acceptable we can proceed with the change.

Reply:
@yushwang @MikhailTryakhov  ping